File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        #using division:
        full_product = math.prod(nums)
        ans = []
    
        for num in  nums:
            if num == 0 and nums.count(0) == 1:
                ans.append(math.prod([x for x in nums if x!= 0]))
            elif num == 0:
                ans.append(0)
            else:
                ans.append(int(full_product / num))
        
        return(ans)
        

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-0.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int[] ans = new int[cost.length];
        for(int i = cost.length -1; i >= 0; i --){
            if (i == cost.length -1){
                ans[i] = cost[i]; 
            }
            else if(i == cost.length - 2){
                ans[i] = cost[i]; 
            }
            else{
                ans[i] = cost[i] + Math.min(ans[i+1], ans[i + 2]);            
            }
            
        }
        
        return(Math.min(ans[0], ans[1]));
        
    }
}
